common/src/main/scala/explore/common/UserPreferencesQueries.scala

```diff
@@ -221,4 +223,73 @@ object UserPreferencesQueries {
       }.void
 
   }
+  @GraphQL(debug = false)
+  object UserTargetPreferencesQuery extends GraphQLOperation[UserPreferencesDB] {
+    val document = """
+      query target_preferences($user_id: String! = "", $target_id: String! = "") {
+        lucuma_target_preferences_by_pk(target_id: $target_id, user_id: $user_id) {
+          fov
+        }
+      }
+    """
+
+    // Gets the layout of a section.
+    // This is coded to return a default in case
+    // there is no data or errors
+    def queryWithDefault[F[_]: MonadError[*[_], Throwable]](
+      uid:         User.Id,
+      tid:         Target.Id,
+      defaultFov:  Angle
+    )(implicit cl: GraphQLClient[F, UserPreferencesDB]): F[Angle] =
+      (for {
+        r <-
+          OptionT
+            .liftF[F, Option[Long]] {
+              query[F](uid.show, tid.show).map { r =>
+                r.lucuma_target_preferences_by_pk.map(_.fov)
+              }
+            }
+            .handleErrorWith(_ => OptionT.none)
+      } yield r.map(Angle.fromMicroarcseconds)).value.map(_.flatten.getOrElse(defaultFov))
+  }
+
+  @GraphQL(debug = false)
+  object UserTargetPreferencesUpsert extends GraphQLOperation[UserPreferencesDB] {
+    val document =
+      """mutation target_preferences_upsert($objects: [lucuma_target_insert_input!]! = {}) {
+        insert_lucuma_target(objects: $objects, on_conflict: {constraint: lucuma_target_pkey, update_columns: target_id}) {
+          affected_rows
+        }
+      }"""
```

Comment:
Just FYI, for cases like this, where the mutation expects an `Array` but we are always going to call it with a single object, we can have the GraphQL mutation build the `Array`.

For example, here the mutation could be declared:
```
mutation target_preferences_upsert($object: lucuma_target_insert_input!) {
  insert_lucuma_target(objects: [$object], ...) {
    ...
  } 
}
```